common/models/user.js

```diff
@@ -558,6 +557,13 @@ module.exports = function(User) {
       return cb.promise;
     }
 
+    try {
+      if (options.password) {
+        UserModel.validatePassword(options.password);
```

Comment:
Here is the documentation explaining how to implement password reset: https://github.com/strongloop/loopback-example-user-management#how-do-you-perform-a-password-reset-for-a-registered-user

Gist - the password is updated via `user.updateAttribute`, which calls `user.updateAttributes` under the hood IIRC.

```js
    User.findById(req.accessToken.userId, function(err, user) {
      if (err) return res.sendStatus(404);
      user.updateAttribute('password', req.body.password, function(err, user) {
```